server/protocols/protocols-imap4/src/test/java/org/apache/james/imapserver/netty/IMAPServerTest.java

```diff
@@ -397,6 +397,76 @@ void largeAppendsShouldBeRejected() {
         }
     }
 
+    @Nested
+    class PlainAuthDisallowed {
+        IMAPServer imapServer;
+        private int port;
+
+        @BeforeEach
+        void beforeEach() throws Exception {
+            imapServer = createImapServer("imapServerPlainAuthDisallowed.xml");
+            port = imapServer.getListenAddresses().get(0).getPort();
+        }
+
+        @AfterEach
+        void tearDown() {
+            imapServer.destroy();
+        }
+
+        @Test
+        void loginShouldFailOnUnEncryptedChannel() {
+            assertThatThrownBy(() ->
+                testIMAPClient.connect("127.0.0.1", port)
+                    .login(USER.asString(), USER_PASS))
+                .hasMessage("Login failed");
+        }
+
+        @Test
+        void capabilityShouldNotAdvertiseLoginOnUnEncryptedChannel() throws Exception {
+            testIMAPClient.connect("127.0.0.1", port);
+
+
+            assertThat(testIMAPClient.capability())
+                .contains("LOGINDISABLED")
+                .doesNotContain("AUTH=PLAIN");
+        }
+    }
+
+    @Nested
+    class PlainAuthDisallowedSSL {
```

Comment:
Isn't that nested class the exact same as the one above? I don't see the SSL difference here... Shouldn't you test the PlainAuthAllowed isntead?